packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -302,16 +309,29 @@ export class EngineReportingAgent<TContext = any> {
     if (protobufError) {
       throw new Error(`Error encoding trace: ${protobufError}`);
     }
-    const encodedTrace = Trace.encode(trace).finish();
 
-    const signature = await this.getTraceSignature({
+    const { signature, graphqlValidationFailure } = await this.getTraceSignature({
       queryHash,
       documentAST,
       queryString,
       operationName,
     });
 
-    const statsReportKey = `# ${operationName || '-'}\n${signature}`;
+    const operationNameOrStatic = graphqlValidationFailure ?
+      validationErrorStaticIdentifier : operationName;
```

Comment:
It seems like we lose the operation name in this process, even if the special flag is set. Is that intentional?

Don't forget that one reason for validation failure is incorrect operation name!

Reply:
That's a good point! I added a field to proto for it.